Manage bookmarks

benoitblanc opened the pull request:
Hi @manisandro 

I have just finished the work to manage bookmarks as discussed in https://github.com/qwc-services/qwc-permalink-service/pull/5

To do so, I have created a new menu item that sould be available only for logged-in users. That is why I created a new `requireAuth` option in menu items for the configuration. Then, I created a plugin and a window to list bookmarks for currently logged-in user. Here are some screenshots :

* Bookmarks are available only for logged-in users : 
![image](https://user-images.githubusercontent.com/8960009/115531007-94c91e00-a294-11eb-9100-75e3036b8932.png) ![image](https://user-images.githubusercontent.com/8960009/115531144-baeebe00-a294-11eb-9e5c-472e51ae478a.png)

* Bookmarks are listed in the window. User can add a new one with a description using the current map :
![image](https://user-images.githubusercontent.com/8960009/115531415-0b661b80-a295-11eb-9c9a-95b2220c7abf.png) 
![image](https://user-images.githubusercontent.com/8960009/115531571-394b6000-a295-11eb-9e7e-facda4086342.png)

* User can open bookmarks in new tab, overwrite bookmark with the current map, or remove bookmark.
![image](https://user-images.githubusercontent.com/8960009/115531918-934c2580-a295-11eb-9915-19e5c9753a11.png)

This plugin uses the work done in https://github.com/qwc-services/qwc-permalink-service/pull/5 and https://github.com/qwc-services/qwc-config-db/pull/2 to create, get, remove bookmarks in QWC configuration database.

Please let me know if you have some remarks on what I have done.

Thank you for your review and your feedback

manisandro commented:
Thanks, looks good at a first glance! Slight nit pick: would it be possible to style the panel a bit more consistently with the other panels? I.e. taking the Share panel as a template for the heading style and padding etc.

benoitblanc commented:
I have changed heading with `h4` and added padding for `div` in the window.

![image](https://user-images.githubusercontent.com/8960009/115674141-a6223100-a34d-11eb-9813-4f0f9c678028.png)

manisandro commented on components/BookmarkWindow.jsx:
```diff
@@ -0,0 +1,102 @@
+/**
+ * Copyright 2016-2021 Sourcepole AG
```
You'll probably want to change the Copyright?

manisandro commented on plugins/Bookmark.jsx:
```diff
@@ -0,0 +1,37 @@
+import ConfigUtils from '../utils/ConfigUtils';
+import BookmarkWindow from '../components/BookmarkWindow';
+ 
+class Bookmark extends React.Component {
```
Is there a reason for creating the wrapping plugin for the component instead of just having the entire component as the plugin in one single file? Also, what do you think about using a sidebar (like the Share plugin) rather than a window?

benoitblanc replied:
@manisandro I think I have created the plugin to be able to enable/disable it in `config.json` file. But I realise it may be useless and I should have the component as the plugin ?

I tried to use a sidebar and a window for my tests. I just preferred the window because it is resizeable and dockable but it can be discussed :)

manisandro replied:
Yes you can spare yourself the boilerplate and just expose the component itself as a plugin.

Personally I'd go with a sidebar for consistency with the other functionalities, but your call.

benoitblanc replied:
@manisandro I did the changes to use component directly as a plugin.

Also, I think you were right about the SideBar, it is more consistent for the user to get all the functionalities at the same place :+1:

manisandro commented:
Thanks!
